### ISSUE TITLE
Allow any key in ifShortcut

### ISSUE THREAD

**Issue by likern** [UltimateHackingKeyboard/agent#1934]
It would be awesome to support range of keys or `any` keyword in `ifShortcut`.

My command is the following
```
ifShortcut timeoutIn 200 29 exec toMouseLayer
// Here I want not only 16 key, but all keys in layer fn4
ifShortcut noConsume timeoutIn 1000 16 final holdLayer fn4 
autoRepeat tapKey space 
```

This (`ifShortcut noConsume timeoutIn 1000 16 final holdLayer fn4 `) is the only what works the way I like and reliably.
So, when I press `space + <some key>` I get key from layer `fn4`, while just space (hitting fast multiple times) and autorepeat space still works.

**Comment by kareltucek** [comment]
I believe you are actually after a secondary role, maybe with some non-trivial resolution strategy. Is that correct?

**Comment by kareltucek** [comment]
Also, why not `holdKey space` instead of that `autoRepeat`?

**Comment by likern** [comment]
> 

Kind of. Multiple secondary roles. So space + space would result in switch to mouse layer, space + key would toggle to fn4.

**Comment by likern** [comment]
> 

Thank you, that works. But looks like with `holdKey space` I've found bug here.

**Comment by likern** [comment]
With that macro
```
ifShortcut timeoutIn 200 29 exec toMouseLayer
ifShortcut noConsume timeoutIn 1000 16 final holdLayer fn4
holdKey space
```
sometimes I get `9` instead of `(` on `fn4` layer.
![Снимок экрана от 2022-12-02 22-30-30](https://user-images.githubusercontent.com/842306/205371538-20ab5b6f-4b5a-4d96-968b-7acf39c04630.png)
![Снимок экрана от 2022-12-02 22-33-11](https://user-images.githubusercontent.com/842306/205371656-c40bf88a-d589-404f-bd5d-f21fa4c66332.png)
![Снимок экрана от 2022-12-02 22-34-15](https://user-images.githubusercontent.com/842306/205371824-affe27a1-d1f2-46fc-9e21-3f4b6380f19c.png)

**Comment by kareltucek** [comment]
> Note: Looking at source code I've found that %0 is similar to what I wanted

The `%0` notation is intended to query current state of the queue, not future state of the queue. Please, do not use it as this. 

> sometimes I get 9 instead of ( on fn4 layer.

This looks like your text editor cannot keep up with the fast input. Still, it should not be a problem when tapping a single key. Are you using any hardware or software in between UHK and the text editor? I mean like an RDP session or a KVM switch.

It should go away if you add `set keystrokeDelay 10` into your `$onInit` macro.

**Comment by likern** [comment]
No, keyboard directly attached to usb port of computer.   Native OS (Linux), gedit (simplest text editor), no virtual machine or anything.

I will test `autorepeat tapKey...` too.

**Comment by likern** [comment]
> No, keyboard directly attached to usb port of computer. Native OS (Linux), gedit (simplest text editor), no virtual machine or anything.
> 
> I will test `autorepeat tapKey...` too.

I've tested with `autorepeat ...` and also could caught this bug. I don't think this is somehow related to text editor beeing slow.

I think that's because in UHK there is freeze/delay/gap between "_changing a layer_" and "_apply automatic shift for key_" - so `fn4` layer has already loaded (and at that moment key `9/(` is pressed), while `Shift` has not applied to this key. That should be a bug in UHK logic.

Might be being able to bind directly to `(` (without using `Shift + 9` could solve this problem.

**Comment by kareltucek** [comment]
> I think that's because in UHK there is freeze/delay/gap between "_changing a layer_" and "_apply automatic shift for key_" - so `fn4` layer has already loaded (and at that moment key `9/(` is pressed), while `Shift` has not applied to this key. 

But in your setup, shift is not part of the layer switch. It is direct part of the 9 action. We are deliberately sending two reports so that modifier / scancode order is unambiguous.

Have you tried the keystrokeDelay?

If you share your UserConfig.json, I can give it a test, but I don't think UHK is sending them in wrong order.

> Might be being able to bind directly to `(` (without using `Shift + 9` could solve this problem.

You can't bind `(` directly without using (or editing) host-OS keymap which features scancode exclusively for `(`. Of course, feel free to edit your host-OS keymap...

**Comment by likern** [comment]
> This looks like your text editor cannot keep up with the fast input.
I've tested with terminal and I think you were right. With terminal can't reproduce this bug.
